### PR TITLE
Update distribution tag for Azure Linux 3.0

### PR DESCRIPTION
**Pull request by jslobodzian**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

Mariner 3.0 is "Azure Linux 3".  This first change only updates the distribution tag and the version number.

### CODE REVIEW

File: toolkit/scripts/build_tag.mk [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 ######## BUILD DEFINES ########
 
-DIST_TAG           ?= .cm2
+DIST_TAG           ?= .azl3
```
Comment (mfrw) [inline toolkit/scripts/build_tag.mk:9]: Unrelated comment:
the `l` reads like a `1`

File: toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh [hunk 2]
```diff
@@ -726,7 +726,7 @@ esac
 unset GCC_ARCH
 #	CFLAGS="-O2 -g" \
 #	CXXFLAGS="-O2 -g" \
-# Libffi is causing error building: find: '/usr/src/mariner/BUILDROOT/libffi-3.4.2-1.cm1.x86_64//usr/lib64': No such file or directory
+# Libffi is causing error building: find: '/usr/src/mariner/BUILDROOT/libffi-3.4.2-1.azl3.x86_64//usr/lib64': No such file or directory
```
Comment (anphel31) [inline toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh:729]: this is just an old comment; not necessary to change

File: toolkit/resources/manifests/package/pkggen_core_aarch64.txt
Comment (tobiasb-ms) [inline toolkit/resources/manifests/package/pkggen_core_aarch64.txt:1]: Looks like this might have a merge conflict in it still....

**Review by tobiasb-ms — CHANGES_REQUESTED** [review]

**Review by tobiasb-ms — APPROVED** [review]

**Comment by RokeJulianLockhart** [comment]
So the project is now Azure Linux 3 officially?

**Pull request merged**